cli/azd/pkg/commands/pipeline/github_provider.go

```diff
@@ -332,6 +332,35 @@ func (p *GitHubCiProvider) configureConnection(
 		if err := ghCli.SetSecret(ctx, repoSlug, "ARM_CLIENT_SECRET", values.ClientSecret); err != nil {
 			return fmt.Errorf("setting terraform env var credentials:: %w", err)
 		}
+
+		// Sets the terraform remote state environment variables in github
+		missingRemoteStateValues := false
+		var terraformConfigError error
+		remoteStateKeys := []string{"RS_RESOURCE_GROUP", "RS_STORAGE_ACCOUNT", "RS_CONTAINER_NAME"}
+		for _, key := range remoteStateKeys {
+			value, ok := azdEnvironment.Values[key]
+			if ok {
+				if err := ghCli.SetSecret(ctx, repoSlug, key, value); err != nil {
+					terraformConfigError = fmt.Errorf("setting terraform remote state variables:: %w", err)
+					break
+				}
+			} else {
+				missingRemoteStateValues = true
+				break
+			}
+		}
+
+		if terraformConfigError != nil || missingRemoteStateValues {
```

Comment:
We are displaying this warning/error even when there's an error setting the GitHub secret.  

This warning/info should only be displayed when `!ok` right?  (when an expected env var is not set)

Also, should we try to get the env vars from either azd-environment or from system environment?  (as customer could have manually set up terraform `remote state` on their system and azd should use it

Reply:
Fixed.